### PR TITLE
MAP-2890

### PR DESCRIPTION
**Pull request by cjost1988**

### CODE REVIEW

**Review by pskt — APPROVED** [review]

File: src/Middleware/LogMiddleware.php [hunk 1]
```diff
@@ -56,25 +57,22 @@ public function log(): \Closure
...
-                            'response' => $this->normalizer->normalize($response),
-                            'client' => $this->clientName,
-                        ];
-                        $logger->info($message, $context);
+                        $logger->info($message, $this->buildContext($request, $response, $duration));
 
                         return $response;
                     },
-                    function ($reason) use ($logger, $request, $formatter) {
+                    function ($reason) use ($logger, $request, $formatter, &$duration) {
```
Comment (kobelobster) [inline src/Middleware/LogMiddleware.php]: No need for reference.

File: src/Middleware/LogMiddleware.php [hunk 2]
```diff
@@ -56,25 +57,22 @@ public function log(): \Closure
...
         return function (callable $handler) use ($logger, $formatter) {
             return function (RequestInterface $request, array $options) use ($handler, $logger, $formatter) {
+                $duration = null;
+                $options['on_stats'] = function (TransferStats $stats) use (&$duration) {
+                    $duration = $stats->getTransferTime();
+                };
+
                 return $handler($request, $options)->then(
-                    function (ResponseInterface $response) use ($logger, $request, $formatter) {
+                    function (ResponseInterface $response) use ($logger, $request, $formatter, &$duration) {
```
Comment (kobelobster) [inline src/Middleware/LogMiddleware.php]: No need for reference.

**Review by kobelobster — APPROVED** [review]

File: src/Middleware/LogMiddleware.php (hunk 2 above)
Comment (kobelobster) [reply src/Middleware/LogMiddleware.php]: Fixed.

File: src/Middleware/LogMiddleware.php (hunk 1 above)
Comment (kobelobster) [reply src/Middleware/LogMiddleware.php]: Fixed.

**Pull request merged**